### PR TITLE
add PUBLIC_MANAGEDURL_CONNECT to override HOST:PORT assumption

### PR DESCRIPTION
**Pull request by dgrove-oss**
Add optional override to assuming that the managed URL can be
constructed simply by concatenating a HOST and PORT pair as HOST:PORT.
This is needed for deployments where multiple services are hosted
using URL rewriting on a single host.

### CODE REVIEW

**Comment by dgrove-oss** [comment]
See also description in https://github.com/apache/incubator-openwhisk/pull/3125

File: scripts/lua/management/lib/apis.lua [hunk 1]
```diff
@@ -87,7 +89,7 @@ function _M.addAPI(dataStore, decoded, existingAPI)
   basePath = basePath:sub(-1) == '/' and basePath:sub(1, -2) or basePath
   -- Create managedUrl object
   local uuid = existingAPI ~= nil and existingAPI.id or utils.uuid()
-  local managedUrl = utils.concatStrings({"http://", MANAGEDURL_HOST, ":", MANAGEDURL_PORT, "/api/", decoded.tenantId})
+  local managedUrl = utils.concatStrings({"http://", MANAGEDURL_CONNECT, "/api/", decoded.tenantId})
```
Comment (mhamann) [inline scripts/lua/management/lib/apis.lua]: With this change, you also shouldn't assume `http` only.

File: scripts/lua/management/lib/apis.lua [hunk 2]
```diff
@@ -27,6 +27,8 @@ local MANAGEDURL_HOST = os.getenv("PUBLIC_MANAGEDURL_HOST")
 MANAGEDURL_HOST = (MANAGEDURL_HOST ~= nil and MANAGEDURL_HOST ~= '') and MANAGEDURL_HOST or "0.0.0.0"
 local MANAGEDURL_PORT = os.getenv("PUBLIC_MANAGEDURL_PORT")
 MANAGEDURL_PORT = (MANAGEDURL_PORT ~= nil and MANAGEDURL_PORT ~= '') and MANAGEDURL_PORT or "8080"
+local MANAGEDURL_CONNECT = os.getenv("PUBLIC_MANAGEDURL_CONNECT")
```
Comment (mhamann) [inline scripts/lua/management/lib/apis.lua]: I wonder if `PUBLIC_GATEWAY_URL` might be a better name for this.

**Comment by dgrove-oss** [comment]
good points;  thanks.  I'll revise.

**Comment by dgrove-oss** [comment]
Patch updated as suggested,

**Comment by mhamann** [comment]
Thanks. LGTM.

**Pull request merged**